### PR TITLE
Feature/prime 2055 effective license lookup

### PR DESCRIPTION
**Pull request by hogmann**
This PR cannot be merged until the refactor is done.

To test:

- launch app and look at the lookup api call.
- using a database tool adjust the effective date of a license's detail to the future
- refresh the page and investigate that call
- see that it is no longer shown.

### CODE REVIEW

**Review by AMiKaH — APPROVED** [review]
LGTM.

Not sure if this applies to backend, but I recall a discussion about using the British spelling for **Licence** instead of **License**

**Review by Paahn — APPROVED** [review]
LGTM

**Review by james-hollinger — APPROVED** [review]
LGTM

File: prime-dotnet-webapi-tests/Unit Tests/LookupService.Tests.cs [hunk 1]
```diff
@@ -57,5 +58,48 @@ public async void TestGetLookups()
...
+
+            var newResults = await service.GetLookupsAsync();
+
+            Assert.NotNull(newResults);
+            Assert.NotEmpty(newResults.Licenses);
+            Assert.Equal(new LicenseConfiguration().SeedData.Count() + 1, newResults.Licenses.Count);
+            Assert.Empty(newResults.Licenses.Where(l => l.Code == future.Code));
+            Assert.NotEmpty(newResults.Licenses.Where(l => l.Code == past.Code));
+
+            // Clean Up Test Data
```
Comment (james-hollinger) [inline prime-dotnet-webapi-tests/Unit Tests/LookupService.Tests.cs:96]: You should not have to do this at the end of your test like this. I'm pretty sure the records wont be there in the next test as is. If records _are_ being left behind, we need a global solution, not having to remove the records at the end of each test like this.

Comment (hogmann) [reply prime-dotnet-webapi-tests/Unit Tests/LookupService.Tests.cs:96]: ahh true, i though i tried it and got an error but must have been something else.

**Review by neophyte57 — CHANGES_REQUESTED** [review]
Is there a reason the Jira ticket says "Effective October 1 2019" whereas `LicenseDetailConfiguration` uses 2019-09-16?  Does it matter?  @itchison

**Comment by itchison** [comment]
> Is there a reason the Jira ticket says "Effective October 1 2019" whereas `LicenseDetailConfiguration` uses 2019-09-16? Does it matter? @itchison

I dont think this matters - Just happy that a date in the past was used. Although, I am curious where this date came from.

**Comment by hogmann** [comment]
> > Is there a reason the Jira ticket says "Effective October 1 2019" whereas `LicenseDetailConfiguration` uses 2019-09-16? Does it matter? @itchison
> 
> I dont think this matters - Just happy that a date in the past was used. Although, I am curious where this date came from.

It is the seeding date we use for any of the configuration files, was what was decided on way back when and stored as a constant for configurations

**Review by neophyte57 — APPROVED** [review]
LGTM

**Review by james-hollinger — APPROVED** [review]
LGTM

**Pull request merged**